### ISSUE TITLE
about Automated Feature Engineering Analysis

### ISSUE THREAD

**Issue by lijm-Au** [LAMDASZ-ML/TabFSBench#2]
Your work is quite thorough and insightful!I hope to gain a deeper understanding of the specific implementation method of 'Feature Generation' in the 'Automated Feature Engineering Analysis' section. Could you please answer the following questions:

- Do the generated new features need to be applied to the training set, and does the model need to be retrained based on the updated training set?

- Does the generation of these new features depend on the known feature missing situations in the training set and test set (that is, are new features generated in a targeted manner based on which features are missing)?

**Comment by ClementCheng0217** [comment]
Thank you for your interest in our work!

Let us clarify the points about 'Automated Feature Engineering Analysis' section you mentioned.

- We elaborate further on the experimental setup for “Feature Generation”  in the 'Automated Feature Engineering Analysis' section: 

  - Firstly, we employed OpenFE, CAAFE, and OcTree to generate features for the heart dataset, with the number of generated features equalling that of the original heart dataset.
  - Secondly, we conducted evaluations on random shift, where the shift feature only comprised the original heart features, and the newly generated features exhibited no shifts.
  - This is why even when the number of features shifted is substantial, the degradation in model performance remains relatively modest (as shown in Table 10).

- Therefore, regarding the question you have raised, our answers are below:

  - The generated new features need to be applied to the training set, and the model needs to be retrained based on the updated training set.
  - The generation of new features doesn't depend on the known feature missing situations in the training set or test set. They are only based on the original training set.

- However, we posit that the targeted generation of new features, based on identifying which specific features have shifted, is worthy of research.

We hope our response has addressed your questions. Please feel free to let us know if you have any other questions!
